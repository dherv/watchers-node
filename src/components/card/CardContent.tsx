import React from "react";
import styled, { ThemeProvider } from "styled-components";

const CardContent = ({ title, theme }: { title: string; theme: any }) => (
  <ThemeProvider theme={theme}>
    <Container>
      <Title>{title}</Title>
      <ScoreContainer>
        <Metacritic>
          <span>Metacritic</span>
          <span>89</span>
        </Metacritic>
        <IMDb>
          <span>IMDb</span>
          <span>8.9</span>
        </IMDb>
      </ScoreContainer>
    </Container>
  </ThemeProvider>
);

const Container = styled.div`
  padding: 2rem 1rem;
  background-color: #323534;
  border-radius: 0 0 10px 10px;
`;

const Title = styled.h4`
  font-weight: 600;
  line-height: 1.5em;
  height: 3em;
  overflow: hidden;
  font-size: ${props => props.theme.fontSize};
`;

const ScoreContainer = styled.div`
  display: ${props => props.theme.textDisplay};
  margin-top: 1rem;
`;

const Score = styled.div`
  font-size: 12px;
  margin-right: 1rem;
  span:first-child {
    font-weight: 600;
    margin-right: 8px;
  }
`;

const Metacritic = styled(Score)``;
const IMDb = styled(Score)``;

export { Score, ScoreContainer, Metacritic, IMDb, Container, Title };

export default CardContent;
