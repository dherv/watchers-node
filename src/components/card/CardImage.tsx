import React from "react";
import styled from "styled-components";
import CardRating from "./CardRating";

const CardImage = ({
  src,
  rating,
  title
}: {
  src: string;
  rating: number;
  title: string;
}) => (
  <Container>
    <Image src={src} alt={`${title} poster`}></Image>
    <CardRating size={48} rating={rating} />
  </Container>
);

const Container = styled.div`
  position: relative;
  border-radius: 10px 10px 0 0;
`;

const Image = styled.img`
  border-radius: 10px 10px 0 0;
`;

export { Container, Image };
export default CardImage;
