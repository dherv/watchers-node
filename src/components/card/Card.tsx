import React from "react";
import CardImage from "./CardImage";
import CardContent from "./CardContent";
import styled from "styled-components";
import { IMovie } from "../../interfaces/Movie";

const Card = ({ movie }: { movie: IMovie }) => {
  const poster_url = `https://image.tmdb.org/t/p/w500${movie.poster_path}`;
  return (
    <Container>
      <CardImage
        src={poster_url}
        rating={movie.vote_average}
        title={movie.original_title}
      ></CardImage>
      <CardContent title={movie.original_title}></CardContent>
    </Container>
  );
};

export const Container = styled.div`
  min-width: 300px;
  margin-right: 4rem;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 10px;
`;

export default Card;
