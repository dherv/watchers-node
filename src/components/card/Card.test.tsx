import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Card from "./Card";
import CardImage from "./CardImage";
import CardContent from "./CardContent";

describe("Card", () => {
  const movie = {
    popularity: 188.965,
    vote_count: 901,
    video: false,
    poster_path: "/ePXuKdXZuJx8hHMNr2yM4jY2L7Z.jpg",
    id: 559969,
    adult: false,
    backdrop_path: "/uLXK1LQM28XovWHPao3ViTeggXA.jpg",
    original_language: "en",
    original_title: "El Camino: A Breaking Bad Movie",
    genre_ids: [80, 18, 53],
    title: "El Camino: A Breaking Bad Movie",
    vote_average: 7.2,
    overview:
      "In the wake of his dramatic escape from captivity, Jesse Pinkman must come to terms with his past in order to forge some kind of future.",
    release_date: "2019-10-11"
  };
  const props = { movie };
  const wrapper = shallow(<Card {...props} />);

  describe("Card component", () => {
    describe("user interface", () => {
      test("should display one CardImageContainer", () => {
        expect(wrapper.find(CardImage)).toHaveLength(1);
      });
      test("should display one CardContent", () => {
        expect(wrapper.find(CardContent)).toHaveLength(1);
      });
    });
    describe("events", () => {});
  });

  describe("snapshot", () => {
    test("should match", () => {
      expect.assertions(1);
      const tree = renderer.create(<Card {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
