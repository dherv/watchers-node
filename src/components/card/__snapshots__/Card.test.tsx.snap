// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot should match 1`] = `
.c3 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 5%;
  z-index: 10;
}

.c4 {
  position: relative;
  top: calc(20% - 67.88225099390857px / 2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 67.88225099390857px;
  height: 67.88225099390857px;
}

.c5 {
  position: relative;
  width: 48px;
  height: 48px;
  background-color: #2e8e89;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  z-index: 1;
}

.c6 {
  position: absolute;
  font-weight: 800;
  font-size: 1.2rem;
  color: #ececec;
  z-index: 10;
}

.c7 {
  position: absolute;
  width: 4px;
  background-color: var(--main-color-text);
  opacity: 0.8;
  z-index: 0;
  top: 0;
  left: 20%;
  height: 20%;
}

.c8 {
  position: absolute;
  width: 4px;
  background-color: var(--main-color-text);
  opacity: 0.8;
  z-index: 0;
  bottom: 0;
  right: 25%;
  height: 80%;
}

.c1 {
  position: relative;
  border-radius: 10px 10px 0 0;
}

.c2 {
  border-radius: 10px 10px 0 0;
}

.c9 {
  padding: 2rem 1rem;
  background-color: #323534;
  border-radius: 0 0 10px 10px;
}

.c10 {
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 1.5em;
  height: 3em;
  overflow: hidden;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 1rem;
}

.c12 {
  font-size: 12px;
  margin-right: 1rem;
}

.c12 span:first-child {
  font-weight: 600;
  margin-right: 8px;
}

.c13 {
  font-size: 12px;
  margin-right: 1rem;
}

.c13 span:first-child {
  font-weight: 600;
  margin-right: 8px;
}

.c0 {
  min-width: 300px;
  margin-right: 4rem;
  box-shadow: 0px 4px 4px rgba(0,0,0,0.25);
  border-radius: 10px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt="El Camino: A Breaking Bad Movie poster"
      className="c2"
      src="https://image.tmdb.org/t/p/w500/ePXuKdXZuJx8hHMNr2yM4jY2L7Z.jpg"
    />
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <div
          className="c5"
          size={48}
        />
        <h5
          className="c6"
        >
          7.2
        </h5>
      </div>
      <div
        className="c7"
      />
      <div
        className="c8"
      />
    </div>
  </div>
  <div
    className="c9"
  >
    <h4
      className="c10"
    >
      El Camino: A Breaking Bad Movie
    </h4>
    <div
      className="c11"
    >
      <div
        className="c12"
      >
        <span>
          Metacritic
        </span>
        <span>
          89
        </span>
      </div>
      <div
        className="c13"
      >
        <span>
          IMDb
        </span>
        <span>
          8.9
        </span>
      </div>
    </div>
  </div>
</div>
`;
